src/charm.py

```diff
@@ -3,61 +3,77 @@
 import sys
 sys.path.append('lib')
 
-from ops.charm import CharmBase
-from ops.framework import StoredState
+from ops.charm import CharmBase, CharmEvents
+from ops.framework import EventBase, EventSource, StoredState
 from ops.main import main
 from ops.model import (
     ActiveStatus,
-    BlockedStatus,
-    WaitingStatus,
     MaintenanceStatus,
+    WaitingStatus,
 )
 
-from interfaces import (
-    HTTPInterfaceProvides,
-    MySQLInterfaceRequires,
-)
-from resources import OCIImageResource
+from interface_http import HTTPServer
+from interface_mysql import MySQLServer, MySQLServerError
+from oci_image import OCIImageResource, FetchError
+
+
+class ConfigurePodEvent(EventBase):
+    pass
+
+
+class ServeWebsiteEvent(EventBase):
```

Comment:
We have independent event types in the base framework because we need to be able to evolve the framework without breaking people's existing assumptions. In this case, we can have a single event type named, for example, "GitLabEvent", and reuse this for every event until the type matters. Once a type is necessary, just change it and nothing will break since this is the charm and not a library.